Fix and re-enable Jupyter Notebook shortcut on Windows

martyngigg opened the pull request:
**Note: This has been based on a branch forms a part of ~~#33889~~**

**Description of work.**

The NSIS script for the new package based on conda-build (#33742) had disabled the notebook shortcut as it was not working.

This fixes the command and also modernizes it to use jupyter and not IPython. This removes a deprecation notice.

This also fixes the broken Windows icons for desktop and start menu shortcuts.

**To test:**

See #33742 and build a standalone package having checked out this branch. Note that it does not matter if the conda channel is set to `nightly`

Fixes #33812
Fixes #33869 

*This does not require release notes* because **there should not be a change from a users perspective**

<!-- Ensure the base of this PR is correct (e.g. release-next or master)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

martyngigg committed:
Archive conda-build artifacts from env subdirectories

martyngigg committed:
Use builder account for conda push

martyngigg committed:
Run script with exit on error set

martyngigg committed:
Avoid early exit on conda-recipes update

Running with bash -e causes git diff --exit-code to stop the script. Inline
the check inside the if to avoid this.

martyngigg committed:
Temporarily disable build and test

Attempting to test conda-build in isolation

sf1919 committed:
Changes to stop failures

- Add missing algorithm headers
- set workspace directory to standalone package build

martyngigg committed:
Make archive artifact paths relative

The archive plugin requires relative paths when archiving artifacts from a build.

martyngigg committed:
Add string header for non-precompiled header build

martyngigg committed:
Add more headers to compile with precompiled headers=off

martyngigg committed:
Use conda-bld path in root of workspace

Avoids long path issues

martyngigg committed:
Add timestamps to Jenkins pipeline output

martyngigg committed:
Add comment on variables expected

thomashampson committed:
Add check to startup script to avoid starting processes recursively

thomashampson committed:
Add version info to package name

martyngigg committed:
Add qt.conf file with relative paths for standalone Linux tarball

thomashampson committed:
Add missing scientific interface dlls to windows package

thomashampson committed:
Add missing escape character

thomashampson committed:
Fix typo in output message

martyngigg committed:
Copy artifacts from any conda-bld subdirectory

The archive step preserves the subfolder structure of the job so
a flat '*.tar.bz2' will only look in the workspace root.

martyngigg committed:
Remove variable definition and call script directly

The declarative pipelines do not make it trivial to simply define a common variable.
Put back the direct calls.

martyngigg committed:
Reenable build/test jobs in pipeline

martyngigg committed:
Fix and re-enable Jupyter notebook shortcut for Windows package

Shortcut calls python directly rather than pythonw as the console
can be useful in this case

sf1919 committed:
Fixed icons for Windows shortcuts

fixes #33869

thomashampson commented on installers/conda/win/project.nsi:
```diff
@@ -185,16 +186,14 @@ Section "-Core installation"
-        # Mantid Notebook shortcuts are temporarily disabled because they are not working!
-        # CreateShortCut "$SMPROGRAMS\$StartMenuFolder\${MANTIDNOTEBOOK_LINK_NAME}" "cmd.exe" "/C $\"call $INSTDIR\bin\pythonw.exe -m notebook --notebook-dir=%userprofile%$\"" "${NOTEBOOK_ICON}"
+        CreateShortCut "$SMPROGRAMS\$StartMenuFolder\${MANTIDWORKBENCH_LINK_NAME}" "$INSTDIR\bin\MantidWorkbench.exe" "" "$INSTDIR\bin\mantid_workbench.ico"
+        CreateShortCut "$SMPROGRAMS\$StartMenuFolder\${MANTIDNOTEBOOK_LINK_NAME}" "$INSTDIR\bin\python.exe -m jupyter notebook --notebook-dir=%userprofile%" "$INSTDIR\bin\mantid_notebook.ico"
```
CreateShortCut requires 4 string arguments. The shortcut is created if the python arguments are moved to a separate string, but unfortunately it doesn't work :(

thomashampson commented on installers/conda/win/project.nsi:
```diff
@@ -185,16 +186,14 @@ Section "-Core installation"
-    # Mantid Notebook shortcuts are temporarily disabled because they are not working!
-    # CreateShortCut "$DESKTOP\${MANTIDNOTEBOOK_LINK_NAME}" "cmd.exe" "/C $\"call $INSTDIR\bin\pythonw.exe -m notebook --notebook-dir=%userprofile%$\"" "${NOTEBOOK_ICON}"
+    CreateShortCut "$DESKTOP\${MANTIDWORKBENCH_LINK_NAME}" "$INSTDIR\bin\MantidWorkbench.exe" "" "$INSTDIR\bin\mantid_workbench.ico"
+    CreateShortCut "$DESKTOP\${MANTIDNOTEBOOK_LINK_NAME}" "$INSTDIR\bin\python.exe -m jupyter notebook --notebook-dir=%userprofile%" "$INSTDIR\bin\mantid_notebook.ico"
```
Same as above

sf1919 committed:
Changes requested by reviewer

- moved LOWER_CASE_SUFFIX
- Fixed Notebook shortcuts
- Fixed Start locations for Notebook shortcuts
- Removed WORKBENCH_ICON references

martyngigg committed:
Remove extra site-packages in Windows bundle

It appears unnecessary from testing up to this point
and was a carbon copy of the one in bin\Lib.

Also removes eigen content that was installed by
accident (including in the original non-conda
bundle).

martyngigg committed:
Add sitecustomize module to Windows bundle

See comments in packaging script for why it
is necessary.

thomashampson reviewed (approved):
This is working correctly now
